src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/BotSegments/Samples/SimpleRandomMovementBehaviour.cs

```diff
@@ -40,8 +40,20 @@ public class SimpleRandomMovementBehaviour : MonoBehaviour
 
         private float _lastPositionTime = float.MinValue;
 
+        private bool _sendingKeysInProgress = false;
+
+        private IEnumerator SendKeys(List<(Key, KeyState)> keyStates)
+        {
+            _sendingKeysInProgress = true;
```

Comment:
This concept seems good, but I don't like it being in user space code.  I feel like we should be able to contain this logic within the KeyboardEventSender.  I'll push a commit with my suggestions